linera-core/src/client/mod.rs

```diff
@@ -439,13 +439,15 @@ struct MultichainTracker {
     tracker: u64,
     // Tracks the highest block height processed for each chain.
     highest_seen: BTreeMap<ChainId, BlockHeight>,
+    other_sender_chains: Vec<ChainId>,
 }
```

Comment:
We could rename this to `SenderChainsTracker`, and add to the doc comment that the chain ID-height pairs belong to chains that sent messages to a chain we are trying to synchronize the inbox of.